server/service/osquery_utils/queries.go

```diff
@@ -30,16 +31,22 @@ type DetailQuery struct {
 	// empty, run on all platforms.
 	Platforms []string
 	// IngestFunc translates a query result into an update to the host struct,
-	// around data that lives on the hosts table.
+	// around data that lives on the hosts table. No other IngestFunc field should
+	// be set.
 	IngestFunc func(ctx context.Context, logger log.Logger, host *fleet.Host, rows []map[string]string) error
 	// DirectIngestFunc gathers results from a query and directly works with the datastore to
 	// persist them. This is usually used for host data that is stored in a separate table.
-	// DirectTaskIngestFunc must not be set if this is set.
+	// No other IngestFunc field should be set.
 	DirectIngestFunc func(ctx context.Context, logger log.Logger, host *fleet.Host, ds fleet.Datastore, rows []map[string]string) error
 	// DirectTaskIngestFunc is similar to DirectIngestFunc except that it uses a task to
 	// ingest the results. This is for ingestion that can be either sync or async.
-	// DirectIngestFunc must not be set if this is set.
+	// No other IngestFunc field should be set.
 	DirectTaskIngestFunc func(ctx context.Context, logger log.Logger, host *fleet.Host, task *async.Task, rows []map[string]string) error
+	// DirectAppConfigIngestFunc is similar to DirectIngestFunc except that it
+	// also receives the current AppConfig as argument. No other IngestFunc field
+	// should be set.
+	// TODO: anyone has better ideas on how to provide the AppConfig to an ingestion?
+	DirectAppConfigIngestFunc func(ctx context.Context, logger log.Logger, host *fleet.Host, appCfg *fleet.AppConfig, ds fleet.Datastore, rows []map[string]string) error
```

Comment:
I don't think it's _too bad_, but at some point we may have too many different ingestion func fields...

Reply:
Do you need to pass the `AppConfig` here? You can get it using `ds.AppConfig()`.